src/os_maild/os_maild_client.c

```diff
@@ -672,3 +612,70 @@ MailMsg *OS_RecvMailQ_JSON(file_queue *fileq, MailConfig *Mail, MailMsg **msg_sm
 
     return NULL;
 }
+
+
+
+void cJSON_PrintTable(cJSON *item, char *printed, size_t *body_size, char *tab, int dep)
+{
+    char *val1 = NULL, *val2 = NULL;
+    int log_size;
+    char *t = malloc(14*sizeof(char));
+    strcpy(t, tab);
+
+    if ((item->type & 0xFF) == cJSON_Number || (item->type & 0xFF) == cJSON_String ||
+        (item->type & 0xFF) == cJSON_False || (item->type & 0xFF) == cJSON_True ||
+        (item->type & 0xFF) == cJSON_Array) {
+
+        val1 = malloc(sizeof(item->string));
+        strcpy(val1, item->string);
+        val1[0] = toupper(val1[0]);
+        val2 = cJSON_PrintUnformatted(item);
+        log_size = strlen(val2) + strlen(val1) + 8;
+
+        if (body_size > log_size) {
+            strncat(printed, tab, strlen(tab));
+            strncat(printed, val1, strlen(val1));
+            strncat(printed, ": ", 2);
+            strncat(printed, val2, body_size);
+            strncat(printed, "\r\n", 4);
+            body_size -= log_size;
+        }
+
+        free(val2);
+        free(val1);
+    }
+    else {
+        if (item->child){
+
+            if (item->string) {
+                log_size = strlen(item->string) + 2;
+
+                if (body_size > log_size) {
+                    val1 = malloc(sizeof(item->string));
+                    strcpy(val1, item->string);
+                    val1[0] = toupper(val1[0]);
+                    strncat(printed, tab, strlen(tab));
+                    strncat(printed, val1, strlen(val1));
+                    strncat(printed, "\n", 2);
+                    body_size -= (2 + strlen(val1));
+                    free(val1);
+                }
+            }
+
+            if(dep < 12){
```

Comment:
`dep` refers to the depth of the search in the incoming JSON? If so, it should not be limited.

Reply:
It's the number of times it's been tabulated.